src/System.Windows.Forms.Design/src/System/ComponentModel/Design/UndoEngine.cs

```diff
@@ -437,11 +411,11 @@ private void OnComponentRemoving(object sender, ComponentEventArgs e)
                 PropertyDescriptorCollection props = TypeDescriptor.GetProperties(comp);
                 foreach (PropertyDescriptor prop in props)
                 {
-                    if (prop.PropertyType.IsAssignableFrom(e.Component.GetType()) &&
+                    if (prop.PropertyType.IsInstanceOfType(e.Component) &&
```

Comment:
Why this change?

Reply:
It felt simpler, but I guess it's unnecessary. I'll rollback

Reply:
Yah. if we do not have full context on why one is better than other,  we should rollback.

Reply:
Okay, I had another look. If I rollback, I have to handle the potential nullability of `e.Component`:

```suggestion
                    if (prop.PropertyType.IsAssignableFrom(e.Component!.GetType() &&
```
But `e.Component` is checked for null a few lines before (line 388), so the new code here is more defensive. Here is the code of `IsInstanceOfType` : 

`public virtual bool IsInstanceOfType([NotNullWhen(true)] object? o) => o == null ? false : IsAssignableFrom(o.GetType());`

So for me the options are:

1. Keep it that way
2. Rollback
3. Rollback and add a manual null check for `e.Component`

What do you think?

Reply:
in this case, `IsInstanceOfType` seems safer to take. Lets keep it.